hugo/content/blog/2021-12-07-new-versioning-scheme/index.md

```diff
@@ -0,0 +1,54 @@
+allows for. We called this the _build number_. This build number was just a number that was incremented with every build of the Mumble binaries that
+we performed. Thus a version could now look like this: 1.4.0.62 and the stable release would then simply use whatever build number was next in the
+queue.
+
+## The problem
+
+In theory that worked great and solved the issue at hand: we now had a unique version for every release (be that a stable or a snapshot release).
+However, as it turns out Windows hasn't arrived in the 21st century yet: it restricts programs to use a three-component versioning scheme. While it is
+possible to set a fourth component, without causing any issues (it is even displayed when checking the program's version), it is ignored for all
+relevant actions such as upgrading to a newer version. However, it seems it wasn't ignored entirely as using a four-component versioning scheme caused
+Windows to install a second (or third, or ...) version of Mumble alongside the already installed version if one was to attempt to upgrade between
+Mumble versions that only differed in the build number.
+
+## The solution
+
+After having researched quite a bit, we came to the conclusion that it seems to be impossible to get Windows to properly respect the build number in
+our version specifier (old dogs don't learn new tricks), so it was clear that we had to find a different solution to this. We evaluated a few
+different possibilities but eventually settled on using the patch version number as a replacement for the build number (which will be dropped). Thus,
+we are back at a 3-component version scheme but now the third component is the build number instead of the patch level (major.minor.build).
+
+This appears to be a purely semantical difference and for the most part, it is. The only notable difference is that now the third component of the
+version will (usually) no longer start at zero and then increment one by one while iterating through the different versions of a stable release
+series. Instead, the build number will be increased with every build we create and thus the build number of the released stable versions will be
+somewhat arbitrary. Therefore, it could be that the first release of the 1.4 series will be e.g. 1.4.53 while the first bug-fix release will then
+become 1.4.281.
+
+This makes it impossible to tell (by just looking at the versions themselves) how many releases lay between two given version numbers. However, in
+most scenarios this is probably not what is interesting anyway. Usually one is only interested in which version is newer and that is still immediately
+obvious by looking at the version numbers.
+
+## Summary
+
+Versions remain in a three component format but the third component will now be the build number instead of the patch level. The latter is no longer

```

Comment:
for clarity
```suggestion
Versions remain in a three component format but the third component will now be the build number instead of the patch level. The patch number is no longer

```

Reply:
Nah, that's repetitive. `latter` should be unambiguous in this context

Reply:
I read it wrong first. I think because when saying "build number instead of patch level", build number is the significant information. The sentence talks about the three components and what the third component is. So I read "the latter" being about the third component, the build number.